### PR TITLE
Add --out-request-list option

### PR DESCRIPTION
**Pull request by danbeam**
- [X] CHANGELOG.md has been updated

Useful for calculating inputs for build tools

### CODE REVIEW

**Comment by justinfagnani** [comment]
We need to make sure this is added to the v3 branch as well.

**Comment by justinfagnani** [comment]
Hmm, looking at what this does, I'm not sure it's really a Vulcanize feature rather than an analyzer feature. Vulcanize just happens to run analysis and know what files are referenced.

**Comment by usergenic** [comment]
@justinfagnani I _think_ this is a bundler feature specifically because it relates to the bundled files instead of just the referenced files.

**Comment by usergenic** [comment]
I.e. it has to honor the flags for the bundler not just analyzer.

**Comment by web-padawan** [comment]
+1 to this feature, I think this might be useful for [webcomponent-bundle-analyzer](https://github.com/samccone/webcomponent-bundle-analyzer)

**Comment by justinfagnani** [comment]
That's the thing, webcomponent-bundle-analyzer just uses the analyzer directly.

**Comment by web-padawan** [comment]
Hmm, you're right. Just noticed that webcomponent-bundle-analyzer doesn't take into account sharding logic at all.

**Comment by danbeam** [comment]
fwiw: I'm thinking --list-urls or --list-requests might be more accurate.  thoughts?

File: bin/vulcanize [hunk 1]
```diff
@@ -126,15 +128,25 @@ args.implicitStrip = !args['no-implicit-strip'];
 args.inlineScripts = args['inline-scripts'];
 args.inlineCss = args['inline-css'];
 
-(new vulcan(args)).process(target, function(err, content) {
+var v = new vulcan(args);
```
Comment (justinfagnani) [inline bin/vulcanize]: let's avoid single-letter variable names, just use `vulcanize`

**Comment by justinfagnani** [comment]
I think maybe `--out-request-list`, since this needs to take a filename, like `--out-html`.

**Comment by danbeam** [comment]
should I be using https://github.com/Polymer/polymer-bundler/tree/1.x as the upstream for this PR now?

**Comment by justinfagnani** [comment]
Yeah, that's the Vulcanize branch

**Comment by justinfagnani** [comment]
@danbeam just in case you didn't know, you don't have to create a new PR to change the base branch, you can just edit the PR and choose a new base.

File: bin/vulcanize [hunk 2]
```diff
@@ -126,15 +128,22 @@ args.implicitStrip = !args['no-implicit-strip'];
...
+var vulcanize = new vulcan(args);
+vulcanize.process(target, function(err, content) {
   if (err) {
     process.stderr.write(require('util').inspect(err));
     process.exit(1);
   }
+
+  if (args['out-request-list']) {
+    var urls = Object.keys(vulcanize.loader.requests).filter(function(url) {
+      return !vulcanize.isExcludedHref(url);
```
Comment (danbeam) [inline bin/vulcanize]: @freshp86 and I felt this was saner overall

**Comment by danbeam** [comment]
p.s. if you need me to squash the commit history I can do that :smile:

**Comment by danbeam** [comment]
Squashed the history.  Is there anybody out there?

**Comment by azakus** [comment]
@usergenic @FredKSchott

**Pull request merged**